core/src/commitment_service.rs

```diff
@@ -120,29 +121,33 @@ impl AggregateCommitmentService {
                 return Ok(());
             }
 
-            let aggregation_data = receiver.recv_timeout(Duration::from_secs(1))?;
-            let aggregation_data = receiver.try_iter().last().unwrap_or(aggregation_data);
-
             let mut aggregate_commitment_time = Measure::start("aggregate-commitment-ms");
-            let update_commitment_slots = {
-                match aggregation_data {
-                    CommitmentAggregationData::TowerCommitmentAggregationData(data) => {
-                        let ancestors = data.bank.status_cache_ancestors();
-                        if ancestors.is_empty() {
-                            continue;
-                        }
-
-                        Self::update_commitment_cache(block_commitment_cache, data, ancestors)
-                    }
-                    CommitmentAggregationData::AlpenglowCommitmentAggregationData(data) => {
-                        Self::alpenglow_update_commitment_cache(
-                            block_commitment_cache,
-                            data.commitment_type,
-                            data.slot,
-                        )
+            let tower_result = receiver.recv_timeout(Duration::from_millis(500));
+            let ag_result = ag_receiver.recv_timeout(Duration::from_millis(500));
```

Comment:
this looks scary. Reduced the timeout in half but will wait on receiving on both. I'm assuming only one of these will typically have messages coming across

Reply:
it's tricky because if one of the channels disconnects we'll end up in a case where we don't receive from the other channel (since the disconnected channel is always ready for `recv`). I'll look into reworking this.

Reply:
fixed here 3f369131a74529e5fc5261d9249ed6d1fcf56e87
The alpenglow sender is hacked to not be dropped when the feature flag is off, when we deal with a migration I'll clean it up properly.